lib/storage/metricsmetadata/store_syntest_test.go

```diff
@@ -0,0 +1,84 @@
+//go:build goexperiment.synctest
```

Comment:
This line could be removed, synctest is generally available since go1.25.  Just make sure the new synctest api is used. 

```
This package was first available in Go 1.24 under GOEXPERIMENT=synctest, with a slightly different API. The experiment has now graduated to general availability. The old API is still present if GOEXPERIMENT=synctest is set, but will be removed in Go 1.26.
```

https://go.dev/doc/go1.25#new-testingsynctest-package